src/components/Dialog/Dialog.js

```diff
@@ -106,86 +121,83 @@ const Actions = styled.div`
  * }
  * ```
  */
-class Dialog extends React.Component {
-  handleActionClick = text => {
-    this.props.onActionClick(text);
-  };
-
-  handleClose = () => {
-    this.props.onClose();
-  };
-
-  render() {
-    const { children, active, actions, overlay } = this.props;
-    return (
-      <Wrapper className={active ? 'weave-dialog active' : 'weave-dialog'}>
-        {overlay && (
-          <Overlay
-            onClick={this.handleClose}
-            className="weave-dialog-overlay"
-          />
-        )}
-        <Window className="weave-dialog-window">
-          <ButtonClose className="weave-dialog-close">
-            <span onClick={this.handleClose} className="fa fa-close" />
-          </ButtonClose>
-          <div className="weave-dialog-content">{children}</div>
-          <Actions className="weave-dialog-actions">
-            {actions &&
-              actions.map((Action, i) => {
-                /* eslint react/no-array-index-key: 0 */
-                if (React.isValidElement(Action)) {
-                  return React.cloneElement(Action, { key: i });
-                }
-                return (
-                  <Button
-                    key={i}
-                    onClick={() => this.handleActionClick(Action)}
-                    text={Action}
-                  />
-                );
-                /* eslint react/no-array-index-key: 0 */
-              })}
-          </Actions>
-        </Window>
-      </Wrapper>
-    );
-  }
-}
+const Dialog = ({
+  active,
+  title,
+  width,
+  actions,
+  onActionClick,
+  onClose,
+  children,
+}) => (
+  <Wrapper active={active}>
+    <Overlay onClick={onClose} />
+    <Window width={width}>
+      <Header>
+        <Title>{title}</Title>
+        <ButtonClose text="" onClick={onClose}>
+          <i className="fa fa-close" />
+        </ButtonClose>
+      </Header>
+      <Content>{children}</Content>
+      <Actions>
+        {actions &&
+          actions.map((Action, index) => {
+            if (React.isValidElement(Action)) {
+              /* eslint react/no-array-index-key: 0 */
+              return React.cloneElement(Action, { key: index });
+              /* eslint react/no-array-index-key: 0 */
```

Comment:
This needs to be set to 2 again to re-enable the rule as an error

Reply:
Good catch! :+1: